File: prob60/tic_tac_toe_9.cs
// 実行時間 20ms
using static System.Console;

internal class Program
{
    static void Main()
    {
        char[] c = new char[5];
        string[] s = new string[5];

        for( int i = 0; i < 5; i++ ) {
            s[i] = ReadLine();
            if( Judge( s[i].ToCharArray() ) == true ) {
                return;
            }
        }
        for( int i = 0; i < 5; i++ ) {
            c = new char[5];
            for( int j = 0; j < 5; j++ ) {
                c[j] = s[j][i];
            }
            if( Judge( c ) == true ) {
                return;
            }
        }
        for( int i = 0; i < 5; i++ ) {
            c[i] = s[i][i];
        }
        if( Judge( c ) == true ) {
            return;
        }
        for( int i = 0; i < 5; i++ ) {
            c[i] = s[i][4 - i];
        }
        if( Judge( c ) == true ) {
            return;
        }
        WriteLine( 'D' );
        return;
    }

    static bool Judge( char[] c )
    {
        bool b = false;
        var sc = System.StringComparison.Ordinal;

        if( string.Compare( string.Join( "", c ), "OOOOO", sc ) == 0 ) {
            b = true;
            WriteLine( 'O' );
        } else if( string.Compare( string.Join( "", c ), "XXXXX", sc ) == 0 ) {
            b = true;
            WriteLine( 'X' );
        }
        return b;
    }
}

File: prob60/tic_tac_toe_9.py
# 実行時間 10ms
def judge(c):
    b = False

    if "".join(c) == "OOOOO":
        b = True
        print("O")
    elif "".join(c) == "XXXXX":
        b = True
        print("X")
    return b

def main():
    s = []
    t = []
    u = []

    for _ in [0] * 5:
        i = input()

        if judge(i) == True:
            return
        s.append(i)
    for i in range(5):
        t.clear()
        for j in range(5):
            t.append(s[j][i])
        if judge(t) == True:
            return
    t.clear()
    for i in range(5):
        t.append(s[i][i])
        u.append(s[i][4 - i])
    if judge(t) or judge(u):
        return
    print("D")

main()
